### PR TITLE
Add API endpoint to SDK /v3/rewards

### PR DESCRIPTION
**Pull request by JohnyArkade**
Task: https://app.clickup.com/t/6931982/OM-4914

Feature
- list rewards
- get reward
- create reward
- update reward
- delete reward

image

### CODE REVIEW

**Comment by kyle-arkade** [comment]
Task linked: [OM-4914 Add API endpoint to SDK /v3/rewards](https://app.clickup.com/t/6931982/OM-4914)

File: package.json [hunk 1]
```diff
@@ -33,14 +33,15 @@
...
     "tsup": "^8.3.0",
     "typescript": "^5.6.3",
-    "vitest": "^2.1.4",
-    "jsonwebtoken": "^9.0.2"
+    "vitest": "^2.1.4"
   },
   "dependencies": {
-    "@types/node": "^18.11.11"
+    "@types/node": "^18.11.11",
+    "luxon": "^3.5.0"
```
Comment (oliverarkade) [inline package.json:44]: Please remove luxon dep

The SDK needs to remain mostly free of external dependencies. 
This is important to keep the SDK un opinionated.

This means the SDK can be loaded anywhere (browser or node) and the users of the SDK aren't forced to install additional dependencies into their project just for an SDK

File: src/tests/integration/omneo/rewards/delete-reward.test.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      issued_at: '2024-12-05 08:30:00',
+      timezone: 'Australia/Melbourne',
+      external_id: getExternalId()
+    }
+    const rewardResponse = await simpleOmneoRequest('POST', '/rewards', rewardPayload).catch((err) => {
+      console.error('SDK list rewards created failed:', err)
+      throw new Error('SDK list rewards created failed')
+    })
+
+    const rewardsRes = await omneo.rewards.list({
```
Comment (oliverarkade) [inline src/tests/integration/omneo/rewards/delete-reward.test.ts:48]: fetch by ID instead and expect a 404 response

File: src/tests/integration/omneo/rewards/list-rewards.test.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import { describe, expect, test, afterAll } from 'vitest'
+import { Omneo } from '../../../../omneo'
+import simpleOmneoRequest from '../../../lib/simple-omneo-request'
+import { Reward, RewardDefinitionCreateInput, RewardResponse } from '../../../../types'
+import { getName, getHandle } from './util'
+import { DateTime } from 'luxon'
```
Comment (oliverarkade) [inline src/tests/integration/omneo/rewards/list-rewards.test.ts:6]: No luxon

File: src/tests/integration/omneo/rewards/update-reward.test.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import { describe, expect, test, afterAll } from 'vitest'
+import { Omneo } from '../../../../omneo'
+import simpleOmneoRequest from '../../../lib/simple-omneo-request'
+import { Reward, RewardDefinitionCreateInput, RewardUpdateInput } from '../../../../types'
+import { getName, getHandle } from './util'
+import { DateTime } from 'luxon'
```
Comment (oliverarkade) [inline src/tests/integration/omneo/rewards/update-reward.test.ts:6]: No luxon

File: src/tests/integration/omneo/rewards/create-reward.test.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import { describe, expect, test, afterAll } from 'vitest'
+import { Omneo } from '../../../../omneo'
+import simpleOmneoRequest from '../../../lib/simple-omneo-request'
+import { RewardDefinitionCreateInput, Reward } from '../../../../types'
+import { getName, getHandle } from './util'
+import { DateTime } from 'luxon'
```
Comment (oliverarkade) [inline src/tests/integration/omneo/rewards/create-reward.test.ts:6]: No luxon

File: src/tests/integration/omneo/rewards/get-reward.test.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
+import { describe, expect, test, afterAll } from 'vitest'
+import { Omneo } from '../../../../omneo'
+import simpleOmneoRequest from '../../../lib/simple-omneo-request'
+import { Reward, RewardDefinitionCreateInput } from '../../../../types'
+import { getName, getHandle } from './util'
+import { DateTime } from 'luxon'
```
Comment (oliverarkade) [inline src/tests/integration/omneo/rewards/get-reward.test.ts:6]: No luxon

**Review by oliverarkade — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (JohnyArkade) [reply package.json]: removed.

File: src/tests/integration/omneo/rewards/update-reward.test.ts (hunk 4 above)
Comment (JohnyArkade) [reply src/tests/integration/omneo/rewards/update-reward.test.ts]: removed.

File: src/tests/integration/omneo/rewards/get-reward.test.ts (hunk 6 above)
Comment (JohnyArkade) [reply src/tests/integration/omneo/rewards/get-reward.test.ts]: removed.

File: src/tests/integration/omneo/rewards/create-reward.test.ts (hunk 5 above)
Comment (JohnyArkade) [reply src/tests/integration/omneo/rewards/create-reward.test.ts]: removed

File: src/tests/integration/omneo/rewards/list-rewards.test.ts (hunk 3 above)
Comment (JohnyArkade) [reply src/tests/integration/omneo/rewards/list-rewards.test.ts]: removed.

File: src/tests/integration/omneo/rewards/delete-reward.test.ts (hunk 2 above)
Comment (JohnyArkade) [reply src/tests/integration/omneo/rewards/delete-reward.test.ts:48]: This is list by external id, so not return 404 response.

File: src/tests/integration/omneo/rewards/delete-reward.test.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+const omneo = new Omneo({
+  tenant: process.env.OMNEO_TENANT as string,
+  token: process.env.OMNEO_TOKEN as string
+})
+const CREATED_REWARDS_DEFINITION_IDS : number[] = []
+const FAILED_REWARDS_IDS : number[] = []
+const testProfileID = process.env.OMNEO_TEST_PROFILE_ID as string
+
+describe('Reward delete', () => {
+  test('SDK Reward delete', async () => {
```
Comment (oliverarkade) [inline src/tests/integration/omneo/rewards/delete-reward.test.ts:16]: This test is testing specifically reward delete.
So the only thing that should be using the SDK is the deletion. 

Everything else should use simpleOmneoRequest

File: src/tests/integration/omneo/rewards/delete-reward.test.ts (hunk 2 above)
Comment (oliverarkade) [reply src/tests/integration/omneo/rewards/delete-reward.test.ts:48]: use simpleOmneoRequest to get by ID
The id is in your `rewardResponse`

You shouldn't use the SDK, and also shouldn't use the list endpoint. 
You should get your specific reward you created

File: src/tests/integration/omneo/rewards/delete-reward.test.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    const { data: rewards } = rewardsRes
+    expect(rewards.length).toBe(1)
+
+    await omneo.rewards.delete(rewardResponse.data.id).catch((err) => {
+      console.error(`SDK Reward delete failed with id:${rewardResponse.data.id}`, err)
+      FAILED_REWARDS_IDS.push(rewardResponse.data.id)
+      throw new Error(`SDK Reward delete failed with id:${rewardResponse.data.id}`)
+    })
+
+    const rewardsRes2 = await omneo.rewards.list({
```
Comment (oliverarkade) [inline src/tests/integration/omneo/rewards/delete-reward.test.ts:60]: dont use SDK here use simpleomneo

File: src/tests/integration/omneo/rewards/list-rewards.test.ts [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+const CREATED_REWARDS_DEFINITION_IDS : number[] = []
+const CREATED_REWARDS_IDS : number[] = []
+const testProfileID = process.env.OMNEO_TEST_PROFILE_ID as string
+
+beforeAll(() => {
+  process.env.TZ = 'Australia/Melbourne'
+})
+
+describe('Rewards list', () => {
+  test('SDK Get Rewards', async () => {
```
Comment (oliverarkade) [inline src/tests/integration/omneo/rewards/list-rewards.test.ts:20]: SDK List Rewards

**Review by oliverarkade — CHANGES_REQUESTED** [review]

**Review by oliverarkade — APPROVED** [review]

**Pull request merged**